src/backend/src/filesystem/ll_operations/ll_mkdir.js

```diff
@@ -31,112 +31,13 @@ class LLMkdir extends LLFilesystemOperation {
     }
 
     async _run () {
-        const { context } = this;
-        const { parent, name } = this.values;
-
-        this.checkpoint('lock requested');
-        this.log.noticeme('GET FSLOCK');
-        const svc_fslock = context.get('services').get('fslock');
-        this.log.noticeme('REQUESTING LOCK', {
-            parent: await parent.get('path'),
-            name,
-        });
-        const lock_handle = await svc_fslock.lock_child(
-            await parent.get('path'),
-            name,
-            MODE_WRITE,
-        );
-        this.log.noticeme('GOT LOCK', {
-            parent: await parent.get('path'),
+        const { parent, name, immutable } = this.values;
+        return parent.provider.mkdir({
```

Comment:
We should await this. Technically it isn't necessary because of Promise chaining, but `await` makes it clearer to the next person reading the code that the method being invoked is async.